lib/browser/api/dialog.js

```diff
@@ -70,196 +50,152 @@ const checkAppInitialized = function () {
-      throw new TypeError('Name field label must be a string')
+    if (options == null) {
+      options = {title: 'Save'}
     }
 
-    if (showsTagField == null) {
-      showsTagField = true
-    }
+    const {settings, wrappedCallback} = setupDialog('save', options)
 
-    const wrappedCallback = typeof callback === 'function' ? function (success, result, bookmarkData) {
-      return success ? callback(result, bookmarkData) : callback()
-    } : null
-    const settings = {title, buttonLabel, defaultPath, filters, message, securityScopedBookmarks, nameFieldLabel, showsTagField, window}
     return binding.showSaveDialog(settings, wrappedCallback)
   },
 
-  showMessageBox: function (...args) {
+  showMessageBox: ({
+      window,
+      buttons = [],
+      cancelId,
+      checkboxLabel = '',
+      checkboxChecked,
+      defaultId = -1,
+      detail = '',
+      icon,
+      normalizeAccessKeys,
+      noLink,
+      message = '',
+      title = '',
+      type = 'none',
+      callback
+    }) => {
     checkAppInitialized()
 
-    let [window, options, callback] = parseArgs(...args)
-
-    if (options == null) {
-      options = {
-        type: 'none'
-      }
-    }
-
-    let {
-      buttons, cancelId, checkboxLabel, checkboxChecked, defaultId, detail,
-      icon, message, title, type
-    } = options
-
-    if (type == null) {
-      type = 'none'
-    }
-
+    // check type value validity
     const messageBoxType = messageBoxTypes.indexOf(type)
     if (messageBoxType === -1) {
       throw new TypeError('Invalid message box type')
     }
-
-    if (buttons == null) {
-      buttons = []
-    } else if (!Array.isArray(buttons)) {
+    if (!Array.isArray(buttons)) {
       throw new TypeError('Buttons must be an array')
     }
-
-    if (options.normalizeAccessKeys) {
-      buttons = buttons.map(normalizeAccessKey)
-    }
-
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
+    if (typeof title !== 'string') {
       throw new TypeError('Title must be a string')
     }
-
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
+    if (typeof message !== 'string') {
       throw new TypeError('Message must be a string')
     }
-
-    if (detail == null) {
-      detail = ''
-    } else if (typeof detail !== 'string') {
+    if (typeof detail !== 'string') {
       throw new TypeError('Detail must be a string')
     }
-
-    checkboxChecked = !!checkboxChecked
-
-    if (checkboxLabel == null) {
-      checkboxLabel = ''
-    } else if (typeof checkboxLabel !== 'string') {
+    if (typeof checkboxLabel !== 'string') {
       throw new TypeError('checkboxLabel must be a string')
     }
 
-    if (icon == null) {
-      icon = null
+    if (normalizeAccessKeys) {
+      buttons = buttons.map(normalizeAccessKey)
     }
 
-    if (defaultId == null) {
-      defaultId = -1
-    }
+    checkboxChecked = !!checkboxChecked
```

Comment:
If we check types of the arguments, should we check type of the `checkboxChecked` argument as well? )